### PR TITLE
Pass Test 25

### PR DESCRIPTION
**Pull request by timzoller**
- combined method _if_all_marbles_in_kennel with the method I started_chose_card_with_Joker. 
- New Method is called _generate_kennel_and_joker_actions and passes additionally Test 25. 
- Methode _if_all_marbles_in_kennel was therefore deleted
- created method _generate_joker_swap_actions to pass Test 26

### CODE REVIEW

**Comment by lukasaebi** [comment]
Which tests did pass?

**Comment by timzoller** [comment]
> Which tests did pass?

25 and 26. when I opened the PR I did not get the list with the tests to mark them..dont know why

**Comment by lukasaebi** [comment]
> > Which tests did pass?
> 
> 25 and 26. when I opened the PR I did not get the list with the tests to mark them..dont know why

Strange. Maybe you can just list them like this:
1, 3-5, ...

**Comment by timzoller** [comment]
I passed following tests: 
1, 3-20, 25, 26, 36-40, 42-49, 51

File: server/py/dog.py [hunk 1]
```diff
@@ -229,24 +233,74 @@ def _get_marbles_in_kennel_and_in_play(self, player: PlayerState) -> tuple[list[
...
 
-    def _if_all_marbles_in_kennel(self, player: PlayerState, marbles_in_kennel: list[Marble]) -> list[Action]:
-        """Lists all actions that are possible if all marbles of a player are in the kennel."""
+    def _generate_kennel_and_joker_actions(self, player: PlayerState, marbles_in_kennel: list[Marble]) -> list[Action]:
+        """
+        Generate possible actions when all marbles are in the kennel and/or
+        when the player has a JOKER card.
+        """
+        actions = []
         start_position = StartNumbers[player.colour]
```
Comment (lukasaebi) [inline server/py/dog.py:244]:
```suggestion
        start_position = StartNumbers[player.colour].value
```

**Comment by lukasaebi** [comment]
Ha grad gmerged wili vo dem Punkt us will witerschaffe, hehe. Top👍

**Pull request merged**